scripts/es-integration-test.sh

```diff
@@ -19,114 +19,88 @@ check_arg() {
 }
 
 setup_es() {
-  local tag=$1
-  local image=docker.elastic.co/elasticsearch/elasticsearch
-  local params=(
-    --detach
-    --publish 9200:9200
-    --env "http.host=0.0.0.0"
-    --env "transport.host=127.0.0.1"
-    --env "xpack.security.enabled=false"
-  )
-  local major_version=${tag%%.*}
-  if (( major_version < 8 )); then
-    params+=(--env "xpack.monitoring.enabled=false")
-  else
-    params+=(--env "xpack.monitoring.collection.enabled=false")
-  fi
-  if (( major_version > 7 )); then
-    params+=(
-      --env "action.destructive_requires_name=false"
-    )
-  fi
-
-  local cid
-  cid=$(docker run "${params[@]}" "${image}:${tag}")
+  local compose_file=$1
+  docker compose -f "${compose_file}" up -d
+  local cid 
+  cid=$(docker compose -f "${compose_file}" ps -q elasticsearch)
   echo "cid=${cid}" >> "$GITHUB_OUTPUT"
-  echo "${cid}"
 }
 
 setup_opensearch() {
-  local image=opensearchproject/opensearch
-  local tag=$1
-  local params=(
-    --detach
-    --publish 9200:9200
-    --env "http.host=0.0.0.0"
-    --env "transport.host=127.0.0.1"
-    --env "plugins.security.disabled=true"
-  )
-  local cid
-  cid=$(docker run "${params[@]}" "${image}:${tag}")
+  local compose_file=$1
+  compose_file="docker-compose/opensearch/v${major_version}.yml"
+  docker compose -f "${compose_file}" up -d
+  local cid 
+  cid=$(docker compose -f "${compose_file}" ps -q opensearch)
   echo "cid=${cid}" >> "$GITHUB_OUTPUT"
-  echo "${cid}"
 }
 
 wait_for_storage() {
   local distro=$1
   local url=$2
-  local cid=$3
+  local compose_file=$3
   local params=(
     --silent
     --output
     /dev/null
     --write-out
     "%{http_code}"
   )
-  local counter=0
-  local max_counter=60
-  while [[ "$(curl "${params[@]}" "${url}")" != "200" && ${counter} -le ${max_counter} ]]; do
-    docker inspect "${cid}" | jq '.[].State'
-    echo "waiting for ${url} to be up..."
+  local max_attempts=60
+  local attempt=0
+  echo "Waiting for ${distro} to be available at ${url}..."
+  until [[ "$(curl "${params[@]}" "${url}")" == "200" ]] || (( attempt >= max_attempts )); do
+    echo "Attempt $(( attempt + 1 )): ${distro} is not yet available at ${url}..."
```

Comment:
can move `attempt=$(( attempt + 1 ))` before this statement and not do math inside the string extrapolation